### PR TITLE
Match calc player damage enemy

### PR DESCRIPTION
**Pull request by ThreeMustyFears**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.17%) on `us`.

✅ This PR fixes 1 warnings!

File: src/19FAF0.c [hunk 1]
```diff
@@ -223,7 +225,739 @@ s32 calc_player_test_enemy(void) {
...
+    s32 isAbilityActive;
+
+    sp20 = FALSE;
+    isFireDamage = FALSE;
+    isElectricDamage = FALSE;
+    isIceDamage = FALSE;
+    tempBinary = FALSE;
+    sp1C = FALSE;
+    wasStatusInflicted = FALSE;
+    sp24 = FALSE;
```
Comment (nanaian) [inline src/19FAF0.c:258]: These can be merged with their declarations above

File: src/19FAF0.c [hunk 2]
```diff
@@ -223,7 +225,739 @@ s32 calc_player_test_enemy(void) {
...
+    target = get_actor(currentTargetID);
+    state = &player->state;
+
+    if (target == NULL) {
+        return 0;
+    }
+
+    targetPart = get_actor_part(target, currentTargetPartID);
+
+    ASSERT (targetPart != NULL);
```
Comment (nanaian) [inline src/19FAF0.c:274]:
```suggestion
    ASSERT(targetPart != NULL);
```

File: src/19FAF0.c [hunk 3]
```diff
@@ -223,7 +225,739 @@ s32 calc_player_test_enemy(void) {
...
+        if (battleStatus->currentAttackElement & DAMAGE_TYPE_WATER) {
+            fx_water_splash(0, state->goalPos.x, state->goalPos.y, state->goalPos.z + 5.0f, 1.0f, 0x18);
+        }
+
+        if (battleStatus->currentAttackElement & DAMAGE_TYPE_ICE) {
+            fx_big_snowflakes(0, state->goalPos.x, state->goalPos.y, state->goalPos.z + 5.0f);
+            isIceDamage = TRUE;
+        }
+
+        isAbilityActive = player_team_is_ability_active(player, ABILITY_ATTACK_FX);
```
Comment (nanaian) [inline src/19FAF0.c:364]: I'd call this var `attackFxType`

**Review by ethteck — APPROVED** [review]

File: src/19FAF0.c [hunk 4]
```diff
@@ -223,7 +225,739 @@ s32 calc_player_test_enemy(void) {
...
+            targetDefense = 0;
+            currentAttackDamage = target->currentHP;
+        }
+
+        if (battleStatus->currentAttackElement & DAMAGE_TYPE_STATUS_ALWAYS_HITS) {
+            targetDefense = 0;
+            currentAttackDamage = 0;
+        }
+
+        if (currentAttackDamage >= 100) {
```
Comment (nanaian) [inline src/19FAF0.c:531]:
```suggestion
        if (currentAttackDamage > 99) {
```
If matches

File: src/19FAF0.c [hunk 5]
```diff
@@ -223,7 +225,739 @@ s32 calc_player_test_enemy(void) {
...
+        currentAttackDamage -= targetDefense;
+
+        if (currentAttackDamage < 0) {
+            currentAttackDamage = 0;
+        }
+
+        if (battleStatus->currentAttackElement & DAMAGE_TYPE_POWER_BOUNCE && currentAttackDamage > 0) {
+            currentAttackDamage += battleStatus->powerBounceCounter;
+
+            if (currentAttackDamage <= 0) {
```
Comment (nanaian) [inline src/19FAF0.c:549]:
```suggestion
            if (currentAttackDamage < 1) {
```

File: src/19FAF0.c [hunk 6]
```diff
@@ -223,7 +225,739 @@ s32 calc_player_test_enemy(void) {
...
+        }
+    }
+
+    if (tempBinary && gBattleStatus.flags1 & BS_FLAGS1_40 || gBattleStatus.flags1 & BS_FLAGS1_40) {
+        if (!(gBattleStatus.flags1 & BS_FLAGS1_80)) {
+            if (player->actorTypeData1[5]) {
+                sfx_play_sound_at_position(player->actorTypeData1[5], 0, state->goalPos.x, state->goalPos.y, state->goalPos.z);
+            }
+
+            if (battleStatus->lastAttackDamage > 0 && (sfx_play_sound(SOUND_231), battleStatus->lastAttackDamage > 0) ||    // TODO remove sfx_play_sound from conditional
```
Comment (ethteck) [inline src/19FAF0.c:834]: did you try doing 

```
if (damage > 0) {
    play_sound
    if (other thing)...
}
```

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/19FAF0.c (hunk 6 above)
Comment (ThreeMustyFears) [reply src/19FAF0.c:834]: Yeah I did buts it's doing two comparisons to damage > 0 in the assembly and I just didn't figure out how to get it to match

Comment (ethteck) [reply src/19FAF0.c:834]: ahh kk, makes sense

File: src/19FAF0.c (hunk 1 above)
Comment (ThreeMustyFears) [reply src/19FAF0.c:258]: I tried that but it causes issues with the order that stack pointers are declared in the assembly

**Review by nanaian — APPROVED** [review]

**Pull request merged**